neo/Network/P2P/ProtocolHandler.cs

```diff
@@ -21,9 +22,32 @@ public class SetVersion { public VersionPayload Version; }
         public class SetVerack { }
         public class SetFilter { public BloomFilter Filter; }
 
+        private class LimitedSet<T>
+        {
+            private int maxCapacity;
+            private OrderedDictionary orderedDictionary;
+
+            public LimitedSet(int maxCapacity = 100)
```

Comment:
How did you reach 100?
Do you think it is enough?

I had thought about a bigger value.

Reply:
I agree with @vncoelho it should hold much more. Maybe like 32K values.

Reply:
It's OK for me as it just lookup most recently hashes, but can increase to a larger size.

Reply:
With such a small size it can defeat its purpose of preventing processing duplicates,  since the cache is easily exhausted with high incoming volume.

Reply:
I just remember that it use to grow very quick.

But 100 looks small because maybe we are requesting many blocks and etc... I am not sure.
Some tests would be good for setting this nicely.

Reply:
👍

Reply:
> I just remember that it use to grow very quick.
> 
> But 100 looks small because maybe we are requesting many blocks and etc... I am not sure.
> Some tests would be good for setting this nicely.

I made the size configurable, you can change it later after some tests.

Reply:
I think we can find a best value and make it fixed.

Reply:
Very pro-active @yongjiema, thanks for this configuration.

It will help during tests.